hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/AvroSchemaEvolutionUtils.java

```diff
@@ -145,8 +146,9 @@ public static Schema reconcileSchemaRequirements(Schema sourceSchema, Schema tar
       return targetSchema;
     }
 
-    InternalSchema sourceInternalSchema = convert(sourceSchema);
     InternalSchema targetInternalSchema = convert(targetSchema);
+    // Use existing fieldIds for consistent field ordering between commits when shouldReorderColumns is true
+    InternalSchema sourceInternalSchema = convert(sourceSchema, shouldReorderColumns ? targetInternalSchema.getNameToPosition() : Collections.emptyMap());
```

Comment:
why only source schema? wny not reorder target schema too?

Reply:
The target schema is the source for the ordering. In this code, the target schema is the existing table and the source is the incoming dataset